db/orders.js

```diff
@@ -0,0 +1,158 @@
+    };
+};
+
+// Does not return order's products
+async function getAllOrders() {
+    try {
+        const {rows} = await client.query(`
+            SELECT o.*,  p.name
+            FROM orders AS o
+            INNER JOIN order_products AS op ON op."orderId" = o.id
+            INNER JOIN products AS p ON op."productId" = p.id;
+        `);
+        return rows;
+    } catch (error) {
+        throw error;
+    };
+};
+
+// Does not return order's products
+async function getOrdersByUser({id}) {
+    try {
+        const {rows: [order]} = await client.query(`
+            SELECT o.*,  p.name
+            FROM orders AS o
+            INNER JOIN order_products AS op ON op."orderId" = o.id
+            INNER JOIN products AS p ON op."productId" = p.id;
+            WHERE "userID" = $1;
+        `, [id])
+    } catch (error) {
+        throw error;
+    };
+};
+
+// Does not return order's products
+async function getOrdersByProduct({id}) {
+    try {
+        const {rows: [order]} = await client.query(`
+            SELECT o.*,  p.name
+            FROM orders AS o
+            INNER JOIN order_products AS op ON op."orderId" = o.id
+            INNER JOIN products AS p ON op."productId" = p.id;
+            WHERE "productId" = $1;
+        `, [id]);
+        return order;
+    } catch (error) {
+        throw error;
+    }
+};
+
+// Does not return order's products
+async function getCartByUser({id}) {
+    try {
+        const{rows: [cart]} = await client.query(`
+            SELECT o.*,  p.name
+            FROM orders AS o
+            INNER JOIN order_products AS op ON op."orderId" = o.id
+            INNER JOIN products AS p ON op."productId" = p.id;
+            WHERE "userID" = $1
+            AND status = "created";
+        `, [id])
+        return cart;
+    } catch (error) {
+        throw error;
+    };
+};
+
+// Does not return order's products
+async function createOrder({status, userID}) {
+    try {
+        const {rows: [order]} = await client.query(`
+            INSERT INTO orders(status, "userID")
+            VALUES ($1, $2)
+            RETURNING *
+            `, [status, userID]);
+        return order;
+    } catch (error) {
+        throw error;
+    }
+};
+
+async function updateOrder({ id, ...fields }) {
+    try {
+        const toUpdate = {};
+        let setStrings = [];
+        let count = 1;
+        for(let column in fields) {
+          if(fields[column] !== undefined) {
+            toUpdate[column] = fields[column];
+            setStrings.push(`"${column}"=$${count}`)
+            count++;
+          };
+        };
+        const setStr = setStrings.join(',');
+        const {rows: [order]} = await client.query(`
+            UPDATE orders 
+            SET ${setStr}
+            WHERE id=${ id }
+            RETURNING *;
+        `, Object.values(toUpdate));
+          return order;
+      } catch (error) {
+        throw error;
+      };
+  };
+
+async function completeOrder({ id }) {
```

Comment:
we could also just call updateOrder with id and status: "completed"

Reply:
Following the milestone guidelines.